internal/provider/inventory_data_source.go

```diff
@@ -4,103 +4,143 @@ import (
+					"If not provided, the inventory will be created in the default organization.",
 			},
-			"hosts": schema.MapNestedAttribute{
-				NestedObject: schema.NestedAttributeObject{
-					Attributes: map[string]schema.Attribute{
-						"hostvars": schema.MapAttribute{
-							ElementType: types.StringType,
-							Computed:    true,
-						},
-					},
-				},
+			"url": schema.StringAttribute{
 				Computed: true,
 			},
+			"name": schema.StringAttribute{
+				Computed: true,
+			},
+			"description": schema.StringAttribute{
+				Computed: true,
+			},
+			"variables": schema.StringAttribute{
+				Optional:   true,
+				CustomType: jsontypes.NormalizedType{},
+			},
 		},
 	}
 }
 
-const ungroupedName string = "ungrouped"
-const allgroupsName string = "all"
+// inventoryDataSourceModel maps the data source schema data.
+type InventoryDataSourceModel struct {
+	Id           types.Int64          `tfsdk:"id"`
+	Organization types.Int64          `tfsdk:"organization"`
+	Url          types.String         `tfsdk:"url"`
+	Name         types.String         `tfsdk:"name"`
+	Description  types.String         `tfsdk:"description"`
+	Variables    jsontypes.Normalized `tfsdk:"variables"`
+}
+
+func (d *InventoryDataSourceModel) ParseHttpResponse(body []byte) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	// Unmarshal the JSON response
+	var apiInventory InventoryAPIModel
+	err := json.Unmarshal(body, &apiInventory)
+	if err != nil {
+		diags.AddError("Error parsing JSON response from AAP", err.Error())
+		return diags
+	}
+
+	// Map response to the inventory datesource schema
+	d.Id = types.Int64Value(apiInventory.Id)
+	d.Organization = types.Int64Value(apiInventory.Organization)
+	d.Url = types.StringValue(apiInventory.Url)
+	d.Name = types.StringValue(apiInventory.Name)
+
+	if apiInventory.Description != "" {
+		d.Description = types.StringValue(apiInventory.Description)
+	} else {
+		d.Description = types.StringNull()
+	}
+	if apiInventory.Variables != "" {
+		d.Variables = jsontypes.NewNormalizedValue(apiInventory.Variables)
+	} else {
+		d.Variables = jsontypes.NewNormalizedNull()
+	}
+	return diags
+}
 
 // Read refreshes the Terraform state with the latest data.
-func (d *inventoryDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
-	var state inventoryDataSourceModel
+func (d *InventoryDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state InventoryDataSourceModel
+	var diags diag.Diagnostics
 
 	// Read Terraform configuration data into the model
 	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
-	hosts, err := d.ReadAnsibleHosts(state.ID.String())
-	if err != nil {
-		resp.Diagnostics.AddError(
-			"Unable to Read Ansible hosts",
-			err.Error(),
-		)
+	url, diags := getURL(state.Url.ValueString(), state.Id.String())
```

Comment:
We can't use `state.Url` here because it won't have a value at this point. Instead the path we provide to `d.client.Get()` should be `"/api/v2/inventories/" + state.Id.String()`.